qa/perf_test.py

```diff
@@ -0,0 +1,487 @@
+            if node["tx_creator"]:
+                txs_creators.append(self.sc_nodes[index])
+            else:
+                non_txs_creator.append(self.sc_nodes[index])
+        return txs_creators, non_txs_creator
+
+    def find_forger_nodes(self):
+        forger_nodes = []
+        for index, node in enumerate(self.sc_nodes_list):
+            if node["forger"]:
+                forger_nodes.append(self.sc_nodes[index])
+        return forger_nodes
+
+    def send_coins_to_multiple_destination(self, utxo_amount, txs_creator_nodes, non_creator_nodes):
+
+        for j in range(len(txs_creator_nodes)):
+            destination_address = non_creator_nodes[0].wallet_createPrivateKey25519()["result"]["proposition"][
+                "publicKey"]
+            for i in range(self.initial_txs):
+                # Populate the mempool - so don't mine a block
+                sendCoinsToAddress(txs_creator_nodes[j], destination_address, utxo_amount, 0)
+                if (i % 1000 == 0):
+                    print("Node " + str(j) + " sent txs: " + str(i))
+            assert_equal(len(allTransactions(txs_creator_nodes[j], False)["transactionIds"]), self.initial_txs)
+            print("Node " + str(j) + " totally sent txs: " + str(self.initial_txs))
+
+    def get_best_node_block_ids(self):
+        block_ids = {}
+        for node in self.sc_nodes:
+            block = http_block_best(node)
+            block_ids[self.sc_nodes.index(node)] = block["id"]
+        print("Block ids: " + str(block_ids))
+        return block_ids
+
+    def find_boxes_of_address(self, boxes, address):
+        address_boxes = []
+        for box in boxes:
+            if box["proposition"]["publicKey"] == address:
+                address_boxes.append(box)
+        return address_boxes
+
+    def log_node_wallet_balances(self):
+        # Output the balance of each node
+        for index, node in enumerate(self.sc_nodes):
+            wallet_boxes = http_wallet_allBoxesOfType(node, "ZenBox")
+            wallet_balance = 0
+            for box in wallet_boxes:
+                wallet_balance += box["value"]
+            if (self.sc_nodes_list[index]["tx_creator"]):
+                print(f"Node{index} (Transaction Creator) Wallet Balance: {wallet_balance}")
+            else:
+                print(f"Node{index} Wallet Balance: {wallet_balance}")
+
+    def get_node_mined_transactions_by_block_id(self, node, block_id):
+        result = http_block_findById(node, block_id)
+        return result["block"]["sidechainTransactions"]
+
+    def txs_creator_send_to_address(self, utxo_amount, txs_creator_node, non_creator_nodes, tps_test):
+        node_index = self.sc_nodes.index(txs_creator_node)
+        destination_address = non_creator_nodes[0].wallet_createPrivateKey25519()["result"]["proposition"][
+            "publicKey"]
+        start_time = time.time()
+
+        if tps_test:
+            max_tps_per_process = 50
```

Comment:
Please add some comments to explain these.

Reply:
Added a lot of comments to explain the code